frontend/packages/kubevirt-plugin/integration-tests/tests/models/wizard.ts

```diff
@@ -258,7 +288,7 @@ export class Wizard {
       } else {
         await this.addDisk(disk);
       }
-      if (provisionSource?.method === ProvisionSource.DISK && disk.bootable) {
+      if (provisionSource === ProvisionSource.DISK && disk.bootable) {
```

Comment:
`provisionSource === ProvisionSource.DISK` this never return true.

Reply:
Can you please debug what is the value of provisionSource then? Maybe we are passing and setting just a simple `"DISK" string or something wrong elsewhere. Or maybe the bootable prop is set to false in your case?

Reply:
The value of them looks exactly the same, but it should not be the same object, so it never be true. 
I tried to use `provisionSource.value` and `ProvisionSource.DISK.value` but it will have other issues.

Reply:
Both of them returns 
```
ProvisionSource {
  getValue: [Function],
  value: 'PXE',
  getDescription: [Function],
  getSource: [Function],
  description: 'PXE (network boot - adds network interface)',
  source: null
}
ProvisionSource {
  getValue: [Function],
  value: 'Disk',
  getDescription: [Function],
  getSource: [Function],
  description: 'Existing PVC (adds disk)',
  source: null
}
```
It fails at create vm from disk.
```
1) Kubevirt create VM using wizard : ID(CNV-2446) Create VM using Disk.
   Error: Please select the boot source.
       at Wizard.<anonymous> /home/gouyang/console/frontend/packages/kubevirt-plugin/integration-tests/tests/models/wizard.ts:83:13
       at Generator.next <anonymous>
       at fulfilled /home/gouyang/console/frontend/packages/kubevirt-plugin/integration-tests/tests/models/wizard.ts:5:58
       at runMicrotasks <anonymous>
       at processTicksAndRejections internal/process/task_queues.js:97:5
   From: Task: Run it("IDCNV-2446 Create VM using Disk.") in control flow
       at UserContext.<anonymous> /home/gouyang/console/frontend/node_modules/jasminewd2/index.js:94:19
   From asynchronous test: 
   Error
       at Suite.<anonymous> /home/gouyang/console/frontend/packages/kubevirt-plugin/integration-tests/tests/vm.wizard.scenario.ts:69:5
       at Env.<anonymous> /home/gouyang/console/frontend/node_modules/jasmine-fail-fast/dist/jasmine-fail-fast.js:57:26
       at Env.wrapper [as describe] /home/gouyang/console/frontend/node_modules/jasmine-fail-fast/node_modules/lodash/index.js:3592:19
       at Object.<anonymous> /home/gouyang/console/frontend/packages/kubevirt-plugin/integration-tests/tests/vm.wizard.scenario.ts:40:1
       at Module._compile internal/modules/cjs/loader.js:1137:30
       at Module.m._compile /home/gouyang/console/frontend/node_modules/ts-node/src/index.ts:400:23
```



If use `provisionSource.value` and `ProvisionSource.DISK.value`, error see in vmtemplate tests.
```
1) Create VM from Template using wizard : ID(CNV-871) Create VM Template using Container.
   TypeError: Cannot read property 'value' of undefined
       at Wizard.<anonymous> /home/gouyang/console/frontend/packages/kubevirt-plugin/integration-tests/tests/models/wizard.ts:294:27
       at Generator.next <anonymous>
       at fulfilled /home/gouyang/console/frontend/packages/kubevirt-plugin/integration-tests/tests/models/wizard.ts:5:58
       at runMicrotasks <anonymous>
       at processTicksAndRejections internal/process/task_queues.js:97:5
   From: Task: Run it("IDCNV-871 Create VM Template using Container.") in control flow
       at UserContext.<anonymous> /home/gouyang/console/frontend/node_modules/jasminewd2/index.js:94:19
   From asynchronous test: 
   Error
       at Suite.<anonymous> /home/gouyang/console/frontend/packages/kubevirt-plugin/integration-tests/tests/vmtemplate.wizard.scenario.ts:50:5
       at Env.<anonymous> /home/gouyang/console/frontend/node_modules/jasmine-fail-fast/dist/jasmine-fail-fast.js:57:26
       at Env.wrapper [as describe] /home/gouyang/console/frontend/node_modules/jasmine-fail-fast/node_modules/lodash/index.js:3592:19
       at Object.<anonymous> /home/gouyang/console/frontend/packages/kubevirt-plugin/integration-tests/tests/vmtemplate.wizard.scenario.ts:25:1
       at Module._compile internal/modules/cjs/loader.js:1137:30
       at Module.m._compile /home/gouyang/console/frontend/node_modules/ts-node/src/index.ts:400:23
```

Reply:
The failure related to the Disk provision source is not as a result of this PR but changes in the golden images feature. we should see a fix in the next HCO release

Reply:
The main issue is that we are using cloneDeep on the data that we use, which clones the object and effectively changes the reference.

The solution is to echange of cloneDeep method in the integration tests with following custom implementation.

```tsx
import { ObjectEnum } from '@console/shared/src/constants/object-enum';
import * as _ from 'lodash';

export const cloneDeepCustom = (object) => {
  return _.cloneDeepWith(object, (value) => {
    if (value instanceof ObjectEnum) {
      return value;
    }
    return undefined;
  });
};
```

PS please keep it in a separate file to prevent circular dependencies

@irosenzw can you please replace all _.isEqual to plain comparisons?